packages/plugin-git/sources/GitFetcher.ts

```diff
@@ -0,0 +1,58 @@
+import {Fetcher, FetchOptions, MinimalFetchOptions}                         from '@berry/core';
+import {Locator, MessageName}                                               from '@berry/core';
+import {execUtils, miscUtils, scriptUtils, structUtils, tgzUtils}           from '@berry/core';
+import {NodeFS, PortablePath, ppath}                                        from '@berry/fslib';
+
+import {GIT_REGEXP}                                                         from './constants';
+import * as gitUtils                                                        from './gitUtils';
+
+export class GitFetcher implements Fetcher {
+  supports(locator: Locator, opts: MinimalFetchOptions) {
+    if (locator.reference.match(GIT_REGEXP))
```

Comment:
What do you think about using https://github.com/IonicaBizau/git-url-parse?

Reply:
Hmm, I would start simple and keep using the `GIT_REGEXP` 😉 However once we need more info from a git url, it will be nice to give it a try (the number of weekly download on `npm` registry is pretty decent) 👍